micro_sam/sam_annotator/image_series_annotator.py

```diff
@@ -108,21 +135,29 @@ def image_series_annotator(
     state = AnnotatorState()
     state.decoder = decoder
     state.initialize_predictor(
-        image, model_type=model_type, save_path=image_embedding_path,
-        halo=halo, tile_shape=tile_shape, predictor=predictor, ndim=2,
-        precompute_amg_state=precompute_amg_state,
+        image, model_type=model_type, save_path=image_embedding_path, halo=halo, tile_shape=tile_shape,
+        predictor=predictor, ndim=3 if is_volumetric else 2, precompute_amg_state=precompute_amg_state,
         checkpoint_path=checkpoint_path, device=device,
     )
-    state.image_shape = image.shape[:-1] if image.ndim == 3 else image.shape
+    state.image_shape = _get_input_shape(image, is_volumetric)
+
+    if image.ndim == 2 or (image.ndim == 3 and not is_volumetric):  # for mono-channel and RGB images
```

Comment:
This is a bit convoluted. I would go for a simpler condition + raise more indicative errors:
```python
if is_volumetric:
  if image.ndim not in (3, 4):
     raise ValueError(f"Invalid image dimensions for 3d annotator, expect 3 or 4 dimensions, got {image.ndim}")
  annotator = Annotator3d(viewer)
else:
  if image.ndim not in (2, 3):
     raise ValueError(f"Invalid image dimensions for 2d annotator, expect 2 or 3 dimensions, got {image.ndim}")
  annotator = Annotator2d(viewer)
```

Reply:
Yes, I agree. Fixed it!